vm/src/obj/objobject.rs

```diff
@@ -4,13 +4,19 @@ use super::objtype;
 use crate::obj::objproperty::PropertyBuilder;
 use crate::pyobject::{
     AttributeProtocol, DictProtocol, IdProtocol, PyAttributes, PyContext, PyFuncArgs, PyObjectRef,
-    PyRef, PyResult, TypeProtocol,
+    PyRef, PyResult, PyValue, TypeProtocol,
 };
 use crate::vm::VirtualMachine;
 
 #[derive(Clone, Debug)]
 pub struct PyInstance;
 
+impl PyValue for PyInstance {
+    fn required_type(_ctx: &PyContext) -> PyObjectRef {
+        panic!("no specific type for PyInstance, don't type check me")
```

Comment:
Wouldn't this be the `object` type?

Reply:
Oh, would it? I'm not really familiar with how the type object system works, I thought the type of PyInstance would be dependent on the class it's an instance of.

Reply:
This method is basically "what type must a value be (or be a subclass of) for it to be valid to extract this payload from it".